newIDE/app/src/CommandPalette/CommandPalette.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import React from 'react';
+import { I18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Dialog from '@material-ui/core/Dialog';
+import TextField from '@material-ui/core/TextField';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { makeStyles } from '@material-ui/core/styles';
+import CommandsContext from '../CommandPalette/CommandsContext';
+import { type NamedCommand } from '../CommandPalette/CommandManager';
+import { fuzzyOrEmptyFilter } from '../Utils/FuzzyOrEmptyFilter';
+
+/**
+ * Filters options both simply and fuzzy-ly,
+ * prioritizing simple-matched options
+ */
+const filterOptions = (
+  options: Array<NamedCommand>,
+  state: { getOptionLabel: NamedCommand => string, inputValue: string }
+) => {
+  const searchText = state.inputValue.toLowerCase();
+  if (searchText === '') return options;
+
+  const directMatches = [];
+  const fuzzyMatches = [];
+  options.forEach(option => {
+    const optionText = state.getOptionLabel(option).toLowerCase();
+    if (optionText.includes(searchText)) return directMatches.push(option);
+    if (fuzzyOrEmptyFilter(searchText, optionText))
+      return fuzzyMatches.push(option);
+  });
+
+  return [...directMatches, ...fuzzyMatches];
+};
+
+const useStyles = makeStyles({
+  scrollPaper: {
```

Comment:
As it's unusual to use makeStyles in the codebase and to override a Material-UI style, let's add a comment "Show the dialog containing the Command Palette in the top of the screen."